### DISCUSSION TITLE
namesrv前面是否可以绑定LB？？？

### DISCUSSION THREAD

**Discussion by xhrg** [apache/rocketmq#7286] [Ideas]
如果我使用一个LB负载均衡器，转发到3个namesrv节点。这样肯定是有问题的。因为可能注册broker到namesrv-1节点，而读的时候，lb又可能把请求转发到namesrv-2节点。

如果我对namserv进行改造，实现彼此之间同步数据，那这样是否可以使用LB负载到三个节点。这样我在写namesrv的时候，可以使用域名的方式。

集群切换的时候会方便一些。

**Comment by RongtongJin** [comment]
> 如果我使用一个LB负载均衡器，转发到3个namesrv节点。这样肯定是有问题的。因为可能注册broker到namesrv-1节点，而读的时候，lb又可能把请求转发到namesrv-2节点。
> 
> 如果我对namserv进行改造，实现彼此之间同步数据，那这样是否可以使用LB负载到三个节点。这样我在写namesrv的时候，可以使用域名的方式。
> 
> 集群切换的时候会方便一些。

对于客户端来说，可以用LB，对于broker来说，要向每一个nameserver注册，所以要获得全量的nameserver地址，否则路由信息将会不一致。

暂时不会考虑namesrv之间同步数据。

**Reply by xhrg** [reply]
您好，感谢您的回复。  


按照你说的，就是目前的4.x版本中。就算不做任何修改。broker部署的时候，填写每一个namesrv的ip。而生产者和客户端，就目前，也是可以使用lb负载到每一个namesrv的，对吧。

如果按照这个逻辑，也就是broker注册的时候，他会每次遍历所有的namesrv注册，对吧。

-------------------

我们从rocketmq的dashboard中，能够看到消费者的节点ip，如果是这个逻辑，我理解消费者并不是只读操作，也要写。
如果消费者通过lb把自己的ip上报给namesrv1，那么namesrv2和3没得上报信息，这个时候dashboard可能是看不到消费者
